nix/node-env.nix

```
@@ -535,6 +535,8 @@ let
             do
                 file="$(readlink -f "$i")"
                 chmod u+rwx "$file"
+                isScript "$file" || continue
+                dos2unix "$file"
```

I think you can use sed:
```suggestion
                sed -i 's/\r*$//' "$file"
```
Also, just curious, doesn't `patchShebangsAuto` automatically patches these files?

---

I'll test that sed command when I'm back at my computer. That would be a lot lighter for sure

And I don't think `patchShebangs` fixes line-endings but I'd have to go check the code again

---

Okay, I've converted it to sed, making this a less invasive change. Thank you for the recommendation!

---

>And I don't think `patchShebangs` fixes line-endings but I'd have to go check the code again

Sorry, I mean `patchShebangsAuto` automatically patches the interpreters in `$out/bin`, and is added [by default to the fixup phase](https://nixos.org/manual/nixpkgs/stable/#patch-shebangs.sh). So we don't need to call patchShebangs ourselves, and this bit could be something like:
```bash
# Create symlink to the deployed executable folder, if applicable
if [ -d "$out/lib/node_modules/.bin" ]
then
    ln -s $out/lib/node_modules/.bin $out/bin
    find $out/bin -type f -print0 | xargs -0 sed -i 's/\r$//'
fi
```

---

Ah, I misunderstood. I'll try to see if we can get by without it then (though that call predated this PR)

Note that we still want to do `isScript` detection because that sed command can do bad things to binaries :)

---

Yes, it turns out to be unnecessary. I've included removing `patchShebangs` there as a drive-by change, but I would like to avoid making too many unrelated changes in this PR, if that's okay

Does what I pushed look good to you now @marsam?

---

Yes, LGTM. Thanks!